File: shoebill_plugins/PlayerVehiclePlugin/src/main/java/lt/ltrp/player/vehicle/dao/PlayerVehicleDao.java
package lt.ltrp.player.vehicle.dao;

import lt.ltrp.player.vehicle.data.PlayerVehicleMetadata;
import lt.ltrp.data.VehicleCrime;
import lt.ltrp.object.LtrpPlayer;
import lt.ltrp.player.vehicle.object.PlayerVehicle;
import net.gtaun.shoebill.data.AngledLocation;

/**
 * @author Bebras
 *         2016.05.23.
 */
public interface PlayerVehicleDao extends VehicleDao {

    // Player vehicles
    int[] getPlayerVehicles(LtrpPlayer player);
    int[] getArrestedPlayerVehicles(LtrpPlayer player);
    int insert(int modelId, AngledLocation spawnLocation, String license, int color1, int color2, float mileage, float fuel, int ownerId,
               int deaths, String alarm, String lock, int lockCrackTime, int lockPrice, int insurance, int doors, int panels, int lights,
               int tires, float health);
    PlayerVehicle get(int vehicleId);
    void update(PlayerVehicle playerVehicle);
    void delete(PlayerVehicle vehicle);
    int getPlayerVehicleCount(LtrpPlayer player);
    int getPlayerVehicleByLicense(String licensePlate);
    PlayerVehicleMetadata getPlayerVehicleMeta(int vehicleId);

    void setOwner(PlayerVehicle vehicle, LtrpPlayer owner);


    void insertCrime(VehicleCrime crime);

}

File: shoebill_plugins/PlayerVehiclePlugin/src/main/java/lt/ltrp/player/vehicle/dao/PlayerVehicleDao.kt
package lt.ltrp.player.vehicle.dao

import lt.ltrp.player.vehicle.data.PlayerVehicleMetadata
import lt.ltrp.data.VehicleCrime
import lt.ltrp.player.`object`.LtrpPlayer
import lt.ltrp.player.vehicle.`object`.PlayerVehicle
import lt.ltrp.vehicle.dao.VehicleDao
import net.gtaun.shoebill.data.AngledLocation

/**
 * @author Bebras
 * *         2016.05.23.
 */
interface PlayerVehicleDao : VehicleDao {


    // Player vehicles
    fun getPlayerVehicles(player: LtrpPlayer): IntArray

    fun getArrestedPlayerVehicles(player: LtrpPlayer): IntArray
    fun insert(modelId: Int, spawnLocation: AngledLocation, license: String, color1: Int, color2: Int, mileage: Float, fuel: Float, ownerId: Int,
               deaths: Int, alarm: String, lock: String, lockCrackTime: Int, lockPrice: Int, insurance: Int, doors: Int, panels: Int, lights: Int,
               tires: Int, health: Float): Int

    operator fun get(vehicleId: Int): PlayerVehicle
    fun update(playerVehicle: PlayerVehicle)
    fun delete(vehicle: PlayerVehicle)
    fun getPlayerVehicleCount(player: LtrpPlayer): Int
    fun getPlayerVehicleByLicense(licensePlate: String): Int
    fun getPlayerVehicleMeta(vehicleId: Int): PlayerVehicleMetadata

    fun setOwner(vehicle: PlayerVehicle, owner: LtrpPlayer)


    fun insertCrime(crime: VehicleCrime)

}
